hiss/team/views.py

```diff
@@ -68,14 +79,25 @@ def form_valid(self, form: JoinTeamForm):
         return super().form_valid(form)
 
 
-class DetailTeamView(shared_mixins.LoginRequiredAndAppliedMixin, generic.DetailView):
+class DetailTeamView(shared_mixins.UserHasTeamMixin, generic.DetailView):
     """
     Renders a Team if the User is a member.
     """
 
     model = Team
     template_name = "team/team_detail.html"
 
+    def handle_no_permission(self):
+        if not self.request.user.is_authenticated:
+            desired_url = self.request.path_info
+            redirect_url = "/accounts/login/?next=" + desired_url
+            return redirect(redirect_url)
+        if not Application.objects.filter(user=self.request.user.pk):
+            return redirect("/status/")
+        if self.request.user.team is None:
+            return redirect(reverse_lazy("team:join"))
+        return redirect("/status/")
+
```

Comment:
Using the previous comments, please re-write the hard-coded redirects with `reverse_lazy`.